presto-orc/src/test/java/com/facebook/presto/orc/TestStorageOrcFileTailSource.java

```diff
@@ -80,12 +88,124 @@ public void testReadExpectedFooterSize()
+                .setCompression(DwrfProto.CompressionKind.NONE)
+                .setCacheMode(DwrfProto.StripeCacheMode.BOTH)
+                .setCacheSize(100);
+        int tailSize = writeTail(footer, postScript, out);
+        out.close();
+
+        // read the file tail with the disabled "read dwrf stripe cache" feature
+        StorageOrcFileTailSource src = new StorageOrcFileTailSource(tailSize, false);
+        TestingOrcDataSource orcDataSource = new TestingOrcDataSource(createFileOrcDataSource());
+        OrcFileTail orcFileTail = src.getOrcFileTail(orcDataSource, metadataReader, Optional.empty(), false);
+
+        assertEquals(orcFileTail.getMetadataSize(), 0);
+        DwrfProto.Footer actualFooter = readFooter(orcFileTail);
+        assertEquals(actualFooter, footer.build());
+
+        // make sure the stripe cache has not been read
+        assertFalse(orcFileTail.getDwrfStripeCacheData().isPresent());
+        assertEquals(orcDataSource.getReadCount(), 1);
+        DiskRange lastReadRange = orcDataSource.getLastReadRanges().get(0);
+        assertEquals(lastReadRange.getLength(), tailSize);
+    }
+
+    @Test
+    public void testReadDwrfStripeCacheIfEnabled()
+            throws IOException
+    {
+        FileOutputStream out = new FileOutputStream(file.getFile());
+
+        // write a fake stripe cache
+        byte[] stripeCache = new byte[100];
+        for (int i = 0; i < stripeCache.length; i++) {
+            stripeCache[i] = (byte) i;
+        }
+        out.write(stripeCache);
+
+        // write the footer and post script
+        DwrfProto.Footer.Builder footer = DwrfProto.Footer.newBuilder()
+                .addAllStripeCacheOffsets(ImmutableList.of(1, 2, 3));
+        DwrfProto.PostScript.Builder postScript = DwrfProto.PostScript.newBuilder()
+                .setCompression(DwrfProto.CompressionKind.NONE)
+                .setCacheMode(DwrfProto.StripeCacheMode.BOTH)
+                .setCacheSize(stripeCache.length);
+        writeTail(footer, postScript, out);
+        out.close();
+
+        // read the file tail with the enabled "read dwrf stripe cache" feature
+        StorageOrcFileTailSource src = new StorageOrcFileTailSource(FOOTER_READ_SIZE, true);
+        OrcDataSource orcDataSource = createFileOrcDataSource();
+        OrcFileTail orcFileTail = src.getOrcFileTail(orcDataSource, metadataReader, Optional.empty(), false);
+
+        assertEquals(orcFileTail.getMetadataSize(), 0);
+        DwrfProto.Footer actualFooter = readFooter(orcFileTail);
+        assertEquals(actualFooter, footer.build());
+
+        // make sure the stripe cache is loaded correctly
+        assertTrue(orcFileTail.getDwrfStripeCacheData().isPresent());
+        DwrfStripeCacheData dwrfStripeCacheData = orcFileTail.getDwrfStripeCacheData().get();
+        assertEquals(dwrfStripeCacheData.getDwrfStripeCacheMode(), DwrfStripeCacheMode.INDEX_AND_FOOTER);
```

Comment:
static import

Reply:
Done. Replaced all constants and enums in TestStorageOrcFileTailSource with static imports.